File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program134.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number and position from user and toggle that bit. 
//		Return modified number.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	public int iMask,iNo,iPos;
	public int iResult;

	Solution(int iNo,int iPos)
	{
		this.iNo = iNo;
		this.iMask = 0x00000001;
		this.iResult = 0;
		this.iPos = iPos;
	}

	int ToggleBit()
	{

		iMask = iMask<<(iPos-1);
		iResult = iNo ^ iMask;

		return iResult;
	}	
}

class Program
{
	public static void main(String[] args)
	{
		int iRet=0;

		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number: ");
		int iValue = sObj.nextInt(); 

		System.out.println("Enter Position: ");
		int iPos = sObj.nextInt(); 
		
		Solution sObj1 = new Solution(iValue,iPos);
		iRet = sObj1.ToggleBit();

		System.out.println("Modified bit is:"+iRet);
	}	
}
/*
OUTPUT:

Enter number:
10
Enter Position:
3
Modified bit is:14

*/

File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program134.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number and position from user and toggle that bit. 
//		Return modified number.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

typedef unsigned int UINT;

class Solution
{
	public:
		UINT ToggleBit(UINT iValue,int iPos)
		{
			UINT iMask = 0x00000001;
			UINT iResult = 0;

			iMask = iMask<< (iPos-1);
			iResult = iValue ^ iMask;

			return iResult;
		}
};

int main()
{
	UINT iValue=0,uRet=0;
	int iPos=0;

	cout<<"Enter number: ";
	cin>>iValue;

	cout<<"Enter Position: ";
	cin>>iPos;

	Solution *sObj = new Solution();
	uRet = sObj->ToggleBit(iValue,iPos);

	cout<<"Modified number is: "<<uRet<<"\n";
	return 0;
}

/*

OUTPUT:

Enter number: 512
Enter Position: 5
Modified number is: 528

*/




